engine/src/main/java/org/kigalisim/engine/support/RechargeEmissionsRecalcStrategy.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Strategy for recalculating recharge emissions.
+ *
+ * <p>This strategy encapsulates the logic previously found in the
+ * recalcRechargeEmissions method of SingleThreadEngine.</p>
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.kigalisim.engine.support;
+
+import org.kigalisim.engine.Engine;
+import org.kigalisim.engine.SingleThreadEngine;
+import org.kigalisim.engine.number.EngineNumber;
+import org.kigalisim.engine.state.Scope;
+
+/**
+ * Strategy for recalculating recharge emissions.
+ */
+public class RechargeEmissionsRecalcStrategy implements RecalcStrategy {
+
+  private Scope scope;
+
+  /**
+   * Create a new RechargeEmissionsRecalcStrategy.
+   *
+   * @param scope The scope to use for calculations, null to use engine's current scope
+   */
+  public RechargeEmissionsRecalcStrategy(Scope scope) {
+    this.scope = scope;
+  }
+
+  @Override
+  public void execute(Engine target) {
```

Comment:
Is this still required? If not, I would prefer just to have the version of execute with the recalcKit.